zaza/charm_lifecycle/func_test_runner.py

```diff
@@ -31,6 +32,54 @@
 import zaza.utilities.run_report as run_report
 
 
+def failure_report(model_aliases, show_juju_status=False):
+    """Report on apps and units in an error state.
+
+    :param model_aliases: Map of aliases to model names.
+    :type model_aliases: Dict
+    :param show_juju_status: Whether include juju status in the summary.
+    :type show_juju_status: bool
+    """
+    logging.error(model_aliases)
+    error_lines = 20
+    for model_alias, model_name in model_aliases.items():
+        logging.error("Model {} ({})".format(model_alias, model_name))
+        status = zaza.model.get_status(model_name=model_name)
+        erred_apps = []
+        for app in status.applications:
+            if status.applications[app].status.status == 'error':
+                erred_apps.append(app)
+        erred_units = []
+        for app in erred_apps:
+            for uname, ustatus in status.applications[app].units.items():
+                if ustatus.workload_status.status == 'error':
+                    erred_units.append(uname)
+        if erred_apps:
+            logging.error("Applications in error state: {}".format(
+                ','.join(erred_apps)))
+        if erred_units:
+            logging.error("Units in error state: {}".format(
+                ','.join(erred_units)))
+        for unit in erred_units:
+            # libjuju has not implemented debug_log yet so get the log
```

Comment:
Can a libjuju bug be created to track this

Reply:
https://github.com/juju/python-libjuju/issues/447